pygfx/renderers/wgpu/_wgpurenderer.py

```diff
@@ -849,3 +961,116 @@ def _update_sampler(self, resource):
             # compare -> only not-None for comparison samplers!
         )
         resource._wgpu_sampler = resource.rev, sampler
+
+    # Picking
+
+    def get_info_at(self, pos):
+        """Get information about the given window location. The given
+        pos is a 2D point in logical pixels (with the origin at the
+        top-left). Returns a dict with fields:
+
+        * "rgba": The value in the color buffer. All zero's when rendering
+          directly to the screen (bypassing post-processing).
+        * "depth_value": The raw value in the depth buffer (0..1). A
+          depth value of 1.0 means that the background is picked.
+        * "position": The position in world coordinates.
+        * "distance_to_camera": The distance (in world units) from the position
+          to the camera position.
+        """
+
+        # Make pos 0..1, so we can scale it to the render texture
+        if self._logical_size:
+            logical_size = self._logical_size
+        else:
+            logical_size = self._canvas.get_logical_size()
+        float_pos = pos[0] / logical_size[0], pos[1] / logical_size[1]
+
+        can_sample_color = self._render_textures[-1].texture is not None
+
+        # Sample
+        encoder = self._device.create_command_encoder()
+        self._copy_pixel(encoder, self._depth_texture, float_pos, 0)
+        if can_sample_color:
+            self._copy_pixel(encoder, self._render_textures[-1], float_pos, 4)
+        queue = self._device.default_queue
+        queue.submit([encoder.finish()])
+
+        # Collect data from the buffer
+        data = self._pixel_info_buffer.read_data()
+        raw_depth = data[0:4].cast("f")[0]
+
+        # Calculate world coordinates using the camera
+        # todo: how does this hold up when a renderer is used with multiple cameras?
+        camera = self._last_camera
```

Comment:
I don't think it does :) you could let the user provide a camera as an (optional?) argument to make it a little more flexible

Reply:
> Personally I think the picking function should only return the color and depth.

Yeah, that would solve that problem.